src/providers/aws/mod.rs

```diff
@@ -49,29 +50,74 @@ pub struct AwsProvider {
 
 impl AwsProvider {
     pub fn try_new() -> Result<AwsProvider> {
-        let client = retry::Client::try_new()?.return_on_404(true);
+        let mut client = retry::Client::try_new()?.return_on_404(true);
+        let token = AwsProvider::fetch_imdsv2_token(client.clone());
+
+        // If IMDSv2 token is fetched successfully, set the header.
+        // Otherwise, proceed with IMDSv1 mechanism.
+        match token {
+            Ok(t) => {
+                client = client.header(
+                    header::HeaderName::from_bytes(b"X-aws-ec2-metadata-token")
+                        .chain_err(|| "setting header name for aws imdsv2 metadata")?,
+                    header::HeaderValue::from_bytes(t.as_bytes())
+                        .chain_err(|| "setting header value for aws imdsv2 metadata")?,
+                );
+            }
+            Err(err) => {
+                warn!("failed to fetch aws imdsv2 session token with: {}", err);
+            }
+        }
 
         Ok(AwsProvider { client })
     }
 
     #[cfg(test)]
-    fn endpoint_for(key: &str) -> String {
+    #[allow(unused_variables)]
```

Comment:
The warning is because `use_latest` is not used. You can directly silence that by naming the argument as `_use_latest` (see [official docs](https://doc.rust-lang.org/stable/book/ch18-03-pattern-syntax.html#ignoring-an-unused-variable-by-starting-its-name-with-_)).